cmd/transition-storage-class.go

```diff
@@ -85,6 +85,38 @@ func (config *TransitionStorageClassConfigMgr) Add(sc madmin.TransitionStorageCl
 	return nil
 }
 
+func (config *TransitionStorageClassConfigMgr) ListStorageClasses() []madmin.TransitionStorageClassConfig {
+	config.RLock()
+	defer config.RUnlock()
+
+	var storageClasses []madmin.TransitionStorageClassConfig
+	for _, cls := range config.S3 {
+		// This makes a local copy of storage-class config before
+		// passing a reference to it.
+		storageClass := cls
+		storageClasses = append(storageClasses, madmin.TransitionStorageClassConfig{
+			Type: madmin.S3,
+			S3:   &storageClass,
+		})
+	}
+
+	for _, cls := range config.Azure {
+		storageClasses = append(storageClasses, madmin.TransitionStorageClassConfig{
+			Type:  madmin.Azure,
+			Azure: &cls,
```

Comment:
shouldn't you make a local copy here as well?

Reply:
yes, thanks for catching this.